halyard-web/src/main/java/com/netflix/spinnaker/halyard/Main.java

```diff
@@ -18,16 +18,34 @@
 
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.builder.SpringApplicationBuilder;
-import org.springframework.boot.web.support.SpringBootServletInitializer;
+import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @Configuration
-@ComponentScan({"com.netflix.spinnaker.config", "com.netflix.spinnaker.halyard",})
+@ComponentScan(
+    value = {
+      "com.netflix.spinnaker.config",
```

Comment:
Do you know what we're depending on from `com.netflix.spinnaker.config`?  I wonder if we actually even want to more explicitly pull in the few things we need from there, as we may end up breaking halyard if we add further things to clouddriver or front50.

Reply:
I agree that it'd be cleaner to have a whitelist instead of a blacklist in that case, we probably don't use many and it'd make the bean dependency explicit.

@<!-- -->ComponentScan has some flexibility with include and exclude filters but for what we want to do that's bending it too much:
Scanning for everything in `com.netflix.spinnaker.halyard` but only selecting a few classes in `com.netflix.spinnaker.config`. This doesn't seem supported.

The kork and clouddriver @<!-- -->Component beans (anything under `com.netflix.spinnaker.config` really) could be refactored into @<!-- -->Configuration with @Conditionals so it'd be easier to integrate the optionally generated Beans when needed. That seems like the right approach for a library exposing spring beans.

Reply:
That makes sense, I figured there probably wouldn't be a way to do this without refactoring the beans in clouddriver and kork.  (I've in the past been surprised by things that kork brings in.)  But certainly out of scope for this boot2 upgrade.

Reply:
Just committed what I think is a pretty satisfying compromise in a8ae5d1, let me know what you think.

Thanks to @scottfrederick and @jkschneider!

Reply:
That looks great, thanks!